### PR TITLE
misc build system swamp draining

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
See individual commit messages.

### CODE REVIEW

File: po/.gitignore [hunk 1]
```diff
@@ -1,3 +0,0 @@
-Makefile.in.in
```
Comment (martinpitt) [inline po/.gitignore:1]: Ugh, admittedly I wasn't even aware that we had gitignore files in subdirectories. By all means let's please kill them. Thanks!

File: Makefile.am [hunk 2]
```diff
@@ -98,6 +98,17 @@ AM_CPPFLAGS = \
 	-DPACKAGE_SYSCONF_DIR=\""$(sysconfdir)"\" \
 	$(NULL)
 
+AM_CFLAGS = \
+	-Wall \
+	-Werror=strict-prototypes \
```
Comment (martinpitt) [inline Makefile.am:103]: Note to self: Apparently all of them are valid for clang as well, the unit test is happy.

**Review by martinpitt — COMMENTED** [review]
Nice stuff, thanks! Not formally approving yet, as still draft/no-test.

File: Makefile.am (hunk 2 above)
Comment (allisonkarlitskaya) [reply Makefile.am]: This patch is actually broken: it's overridden via the various `*_CFLAGS` variables that we define per-target.  I've dropped this for now, and will bring it back as part of a more comprehensive cleanup that I've been preparing.

**Review by martinpitt — APPROVED** [review]
Thanks!

**Comment by allisonkarlitskaya** [comment]
The journal fails are #17049

**Pull request merged**